docs/spec.md

```diff
@@ -139,26 +139,26 @@ StableHLO programs cannot directly represent values of these types (as a result,
 it is idiomatic to represent scalar values of type `T` with 0-dimensional tensor
 values of type `tensor<T>`).
 
-  * **Boolean type** represents boolean values `true` and `false`.
-  * **Integer types** can be either signed (`si`) or unsigned (`ui`) and have
-    one of the supported bit widths (`4`, `8`, `16`, `32` or `64`).
-    Signed `siN` types represent integer values from `-2^(N-1)` to `2^(N-1)-1`
-    inclusive, and unsigned `uiN` types represent integer values from `0` to
-    `2^N-1` inclusive.
-  * **Floating-point types** can be one of the following:
-    * `f8E4M3FN` and `f8E5M2` types corresponding to respectively the
-    `E4M3` and `E5M2` encodings of the FP8 format described in
-    [FP8 Formats for Deep Learning](https://arxiv.org/abs/2209.05433).
-    * `bf16` type corresponding to the `bfloat16` format described in
-    [BFloat16: The secret to high performance on Cloud TPUs](https://cloud.google.com/blog/products/ai-machine-learning/bfloat16-the-secret-to-high-performance-on-cloud-tpus).
-    * `f16`, `f32` and `f64` types corresponding to respectively
-    `binary16` ("half precision"), `binary32` ("single precision") and
-    `binary64` ("double precision") formats described in
-    [the IEEE 754 standard](https://ieeexplore.ieee.org/document/8766229).
-  * **Complex types** represent complex values that have a **real part**
-    and an **imaginary part** of the same **element type**. Supported complex
-    types are `complex<f32>` (both parts are of type `f32`) and `complex<f64>`
-    (both parts are of type `f64`).
+* **Boolean type** represents boolean values `true` and `false`.
+* **Integer types** can be either signed (`si`) or unsigned (`ui`) and have
+  one of the supported bit widths (`4`, `8`, `16`, `32` or `64`).
+  Signed `siN` types represent integer values from `-2^(N-1)` to `2^(N-1)-1`
+  inclusive, and unsigned `uiN` types represent integer values from `0` to
+  `2^N-1` inclusive.
+* **Floating-point types** can be one of the following:
+  * `f8E4M3FN` and `f8E5M2` types corresponding to respectively the
+  `E4M3` and `E5M2` encodings of the FP8 format described in
+  [FP8 Formats for Deep Learning](https://arxiv.org/abs/2209.05433).
```

Comment:
Re: "In multiline list items, subsequent lines are indented by 2 spaces". In some parts of our Markdown files, e.g. this one, this convention doesn't hold. Do we want to update these files in this PR, or that will be done separately?

Reply:
Ah, that one was a missed fix. I've now indented these lines. Unfortunately, markdownlint does not check for alignment like this, so we just have to make a best-effort. There is also an exception for lists of numbered constraints: there's an unspoken convention to align with the text following the C number. For example:

```
* (C7) $0 \le$ `replica_groups`[i] $\lt$ size(`replica_groups`) $\forall i$
       in `indices(replica_groups)`.
```

I think this is well and good to aid in readability. I think the general idea is to indent *at least* 2 characters, but I worry that explaining this exception would require far too much text. So let's just be reasonable with adherence to the styles that aren't lint checked. :)